fftools/ffmpeg_qsv.c

```diff
@@ -48,6 +48,22 @@ static int qsv_device_init(InputStream *ist)
 {
     int err;
     AVDictionary *dict = NULL;
+    HWDevice *hw_device = NULL;
+
+    if (ist->hwaccel_device) {
+        hw_device = hw_device_get_by_name(ist->hwaccel_device);
+        if (hw_device && hw_device->type == AV_HWDEVICE_TYPE_QSV) {
+            hw_device_ctx = av_buffer_ref(hw_device->device_ref);
+            if (!hw_device_ctx)
+                err = AVERROR(ENOMEM);
+            else
+                err = 0;
+            goto err_out;
```

Comment:
why we need err_out for err = 0

Reply:
Because "err_out" release the resource, I want to make sure the resource is released. However, I found at this line, the resource is not allocated yet, so it is unnecessary. Now I changed it. Thanks.